src/components/pages/blog-post/blog-post.tsx

```diff
@@ -33,114 +56,246 @@ const PanelContainer = styled.div`
   }
 `;
 
-const contentfulRenderOptions: Options = {
-  renderMark: {
-    [MARKS.CODE]: (text) => <code className="language-text">{text}</code>,
-  },
-  renderNode: {
-    [INLINES.HYPERLINK]: (props, children) =>
-      customComponents.a({
-        children,
-        href: props.data.uri,
-        target: '_blank',
-        rel: 'nofollow noopener noreferrer',
+/**
+ * Custom Contentful Rich Text Renderer
+ *
+ * Will automatically filter out ContentfulFootnote elements and appends an
+ * ordered list of footnotes after the rendered rich text.
+ *
+ * @param data rich text data
+ * @returns rendered elements
+ */
+const customContentfulRenderer = (
+  data: RenderRichTextData<ContentfulRichTextGatsbyReference>,
+) => {
+  /**
+   * Filter out footnotes from content references
+   */
+  const footnotes: ContentfulFootnoteReference[] = data.references
+    .filter(
+      (reference) =>
+        reference.__typename === ContentfulCustomModel.CONTENTFUL_FOOTNOTE,
+    )
+    .map((reference) => reference as ContentfulFootnoteReference);
+
+  /**
+   * Create an object that contains information we'll use later
+   */
+  const footnoteReferences: FootnoteReference = {};
+  footnotes.map(
+    (footnote, index) =>
+      (footnoteReferences[footnote.contentful_id] = {
+        index: index + 1,
+        anchor: `footnote-${index + 1}`,
+        referenceAnchor: `footnote-${index + 1}-reference`,
       }),
-    [BLOCKS.HEADING_1]: (props, children) => customComponents.h1({ children }),
-    [BLOCKS.HEADING_2]: (props, children) => customComponents.h2({ children }),
-    [BLOCKS.HEADING_3]: (props, children) => customComponents.h3({ children }),
-    [BLOCKS.HEADING_4]: (props, children) => customComponents.h4({ children }),
-    [BLOCKS.HEADING_5]: (props, children) => customComponents.h5({ children }),
-    [BLOCKS.HEADING_6]: (props, children) => customComponents.h6({ children }),
-    [BLOCKS.HEADING_6]: (props, children) => customComponents.h6({ children }),
-    [BLOCKS.OL_LIST]: (props, children) => customComponents.ol({ children }),
-    [BLOCKS.UL_LIST]: (props, children) => customComponents.ul({ children }),
-    [BLOCKS.QUOTE]: (props, children) =>
```

Comment:
The diff got mixed up and list a lot of removals and additions, but I can easily follow looking directly into the file 👍